Universal/Sniffs/DeclareStatements/DeclareStatementsStyleSniff.php

```diff
@@ -0,0 +1,241 @@
+     *
+     * @since 1.0.0
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [
+            T_DECLARE
+        ];
+    }
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @since 1.0.0
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        $openParenPtr  = $tokens[$stackPtr]['parenthesis_opener'];
+        $closeParenPtr = $tokens[$stackPtr]['parenthesis_closer'];
+
+        if (isset($openParenPtr, $closeParenPtr) === false) {
+            // Parse error or live coding, bow out.
+            return;
+        }
+
+        $directiveStrings = [];
+        // Get the next string and check if it's an allowed directive.
+        // Find all the directive strings inside the declare statement.
+        for ($i = $openParenPtr; $i <= $closeParenPtr; $i++) {
+            if ($tokens[$i]['code'] === \T_STRING) {
+                $directiveStrings[] = $tokens[$i]['content'];
+            }
+        }
+
+        foreach ($directiveStrings as $directiveString) {
+            if (!in_array($directiveString, $this->allowedDirectives, true)) {
+                $phpcsFile->addError(
+                    sprintf(
+                        'Declare directives can be one of: %1$s. "%2$s" found.',
+                        implode(', ', $this->allowedDirectives),
+                        $directiveString
+                    ),
+                    $stackPtr,
+                    'WrongDeclareDirective'
+                );
+                return;
+            }
+            unset($directiveString);
+        }
+
+        // Curly braces.
+        $hasScopeOpenerCloser = isset($tokens[$stackPtr]['scope_opener']);
+
+        // If strict types is defined using curly brace, throw error.
+        if ($hasScopeOpenerCloser !== false && in_array('strict_types', $directiveStrings, true)) {
+            $phpcsFile->addError(
+                sprintf(
+                    'strict_types declaration must not use block mode. Opening brace found on line %d',
+                    $tokens[$stackPtr]['line']
+                ),
+                $stackPtr,
+                'Forbidden'
+            );
+            return;
+        }
+
+        // Fix for the case when the code is between the curly braces for the strict_types.
+        $codePtr = $phpcsFile->findNext(Tokens::$emptyTokens, ($closeParenPtr + 1), null, true);
+
+        /* If the code pointer is not one of:
+         *    \T_SEMICOLON, \T_CLOSE_TAG or \T_OPEN_CURLY_BRACKET, \T_COLON
+         * throw an error.
+         */
```

Comment:
```suggestion
        /*
         * If the code pointer is not one of:
         *    \T_SEMICOLON, \T_CLOSE_TAG or \T_OPEN_CURLY_BRACKET, \T_COLON
         * throw an error.
         */
```

I'm wondering if this comment is actually needed ?